u8timeseries/models/statistics.py

```diff
@@ -0,0 +1,198 @@
+    :param other: The TimeSeries to remove.
+    :param model: The type of model considered (either 'additive' or 'multiplicative').
+    :return: A TimeSeries [new_ts], defined by removing [other] from [ts].
+    """
+    if (other.values() < 1e-6).any() and model == 'multiplicative':
+        print("WARNING indexes equal to zero, cannot remove for a multiplicative model.")
+        return ts
+    else:
+        if model == 'multiplicative':
+            new_ts = ts / other
+        else:
+            new_ts = ts - other
+
+    return new_ts
+
+
+def remove_seasonality(ts: 'TimeSeries', freq: int = None, model: str = 'multiplicative') \
+                                                        -> 'TimeSeries':
+    """
+    Adjusts the TimeSeries [ts] for a seasonality of order [frequency] using the [model] decomposition.
+
+    :param ts: The TimeSeries to adjust.
+    :param freq: The seasonality period to use.
+    :param model: The type of decomposition to use (additive or multiplicative).
+    :return: A tuple (new_ts, seasonality) of TimeSeries where new_ts is the adjusted original TimeSeries and
+            seasonality is a TimeSeries containing the removed seasonality.
+    """
+
+    _, seasonality = extract_trend_and_seasonality(ts, freq, model)
+
+    new_ts = remove_from_series(ts, seasonality, model)
+
+    return new_ts
+
+
+def remove_trend(ts: 'TimeSeries', model='multiplicative') -> 'TimeSeries':
+    """
+    Adjusts the TimeSeries [ts] for a trend using the [model] decomposition.
+
+    :param ts: The TimeSeries to adjust.
+    :param model: The type of decomposition to use (additive or multiplicative).
+    :return: A tuple (new_ts, trend) of TimeSeries where new_ts is the adjusted original TimeSeries and
+            seasonality is a TimeSeries containing the removed trend.
+    """
+    trend, _ = extract_trend_and_seasonality(ts, model=model)
+
+    new_ts = remove_from_series(ts, trend, model)
+
+    return new_ts
+
+
+def plot_acf(ts: 'TimeSeries', m: int = None, max_lag: int = 24, alpha: float = 0.05,
+             fig_size: Tuple[int, int] = (10, 5)):
+    """
+    Plots the ACF of [ts], highlighting it at lag [m], with corresponding significance interval.
+
+    :param ts: The TimeSeries whose ACF should be plotted.
+    :param m:  The lag to highlight.
+    :param max_lag: The maximal lag order to consider.
+    :param alpha: The confidence interval.
+    :param fig_size: The size of the figure to be displayed.
+    :return:
+    """
+    r = acf(ts.values(), nlags=max_lag)
+
+    # Computes the confidence interval at level alpha for all lags.
+    stats = []
+    for i in range(1, max_lag+1):
+        stats.append(_bartlett_formula(r[1:], i, len(ts)))
+
+    plt.figure(figsize=fig_size)
+
+    for i in range(len(r)):
+        plt.plot((i, i), (0, r[i]), color=('red' if m is not None and i == m else 'black'), lw=.5)
+
+    acf_band = [(r[1:].mean() + norm.ppf(1 - alpha/2) * r[1:].var()) * stat for stat in stats]
```

Comment:
```suggestion
    upp_band = r[1:].mean() + norm.ppf(1 - alpha/2) * r[1:].var()
    acf_band = [upp_band * stat for stat in stats]
```
Small tweak to avoid unnecessary computation, especially if workin with big timeseries